ons-spark/spark-concepts/cache.md

```diff
@@ -0,0 +1,591 @@
+
+Most of this list is beyond the scope of this article, but it's useful to know that these factors effect the exact size of a data set.
+
+As usual we'll start with some imports and creating a `SparkSession`.
+````{tabs}
+```{code-tab} py
+from pyspark.sql import SparkSession, functions as F
+import yaml
+
+spark = (SparkSession.builder.master("local[2]")
+         .appName("cache")
+         .getOrCreate())
+```
+
+```{code-tab} r R
+
+library(sparklyr)
+library(dplyr)
+
+sc <- sparklyr::spark_connect(
+    master = "local[2]",
+    app_name = "cache",
+    config = sparklyr::spark_config())
+
+```
+````
+Next we need some data. Here we're using a small amount of data because we're running a local session which has very imited resource. 
+
+Note that `cache()` n PySpark is a transformation, so to initiate moving the data into memory we need an action, hence the `.count()`.
+
+In sparklyr there is a `force=TRUE` argument in the `tbl_cache()` function meaning there is no need to pipe this into a row count. Also note that the name of the DataFrame will appear in the Spark UI, which is a nice feature sparklyr. To do the same in PySpark we would need to register the DataFrame as a temporary table and give it a name.
+````{tabs}
+```{code-tab} py
+with open("../../../config.yaml") as f:
+    config = yaml.safe_load(f)
+
+population = spark.read.parquet(config['population_path'])
+
+population.cache().count()
+```
+
+```{code-tab} r R
+
+config <- yaml::yaml.load_file("ons-spark/config.yaml")
+
+population <- sparklyr::spark_read_parquet(
+    sc,
+    path=config$population_path,
+    name="population")
+
+sparklyr::tbl_cache(sc, "population", force=TRUE)
+
+```
+````
+
+```plaintext
+2297
+```
+Now we can take a look in the Spark UI, which is reached using the URL http://localhost:4040/jobs/. Note this address is used for a local Spark session, for more information on how to navigate to the Spark UI see the [documentation on monitoring](https://spark.apache.org/docs/latest/monitoring.html).
+
+Within the Spark UI, if you were to head over to the *Storage* tab and you would see an RDD stored in memory, in two partitions and using 35.7 kB of executor memory.
+
+![cached DataFrame in memory](../images/cache_memory.png)
+
+You can also click on the RDD name to get more information, such as how many executors are being used and the size of each partition.
+
+![partition details of cached DataFrame](../images/cache_memory_partitions.png)
+
+## Persist
+
+There is another function that can be used to persist DataFrames, `persist()`/`sdf_persist()` in PySpark/sparklyr is a general form of `cache()`/`tbl_cache()`. It is similar to doing a cache but we are able to specify where to store the data, e.g. use memory but allow spill over to executor disk if the executor memory is full. `persist()`/`sdf_persist()` takes a `StorageLevel` argument to specify where to cache the data. Options for storage levels are:  
+
+`MEMORY_ONLY`
```

Comment:
Is it worth elaborating on these very briefly?

I think they'd also be better formatted as a list

Reply:
I think these are mostly self explanatory so will mark as resovled